pkg/controllers/federatedtypeconfig/federatedtypeconfig_controller.go

```diff
@@ -173,6 +173,16 @@ func (c *Controller) Run(stopChan <-chan struct{}) {
 		return
 	}
 
+	if c.controllerConfig.CreateCrdForFtcs {
+		ftcs := c.ftcStore.List()
+		for _, obj := range ftcs {
+			ftc := obj.(*fedcorev1a1.FederatedTypeConfig)
+			if err := c.ensureFederatedObjectCrd(ftc); err != nil {
+				klog.Error(fmt.Errorf("cannot ensure federated object CRD for %q: %w", ftc.Name, err))
```

Comment:
My logic here is flawed. I realized I should not have moved the creation of federated CRDs out of the reconcile function because:

1. We need to handle the creation of new FTCs after the controller-manager has started
2. The error returned might be recoverable (such as timeout when getting from apiserver) and we can use the workerqueue semantics to retry

However, I think panicking is not a good idea because even if we fail to create the federated CRD for a single FTC, there are other controllers and ftcs that can continue to work properly and we should allow them to run.

Reply:
My initial motivation for moving it out of the reconcile function is because of a potential deadlock caused by:

1. Sync controller's dependency on the federatednamespace crd
2. FTC controller starting subcontrollers synchronously

So what might happen:

1. FTC controller receives events for a FTC that are not the namespace FTC
2. All workers are used to process these events
3. Each worker attempts to start the sync controller synchronously
4. Each sync controller waits for the federated namespace cache to sync (but this can never happen because the federated namespace CRD was never created)
5. All workers are blocked

Reply:
The new approach is to only ensure the federated crd for namespaces outside the reconcile function and panic if we fail to do so. This is because the federated namespace is a requirement that will block some of the controllers - namely sync and nsautopropagation.

The federated crd for other resources will continue to be handled in the reconcile function.

cc @gary-lgy @mrlihanbo